### PR TITLE
refactor(tests): reorganize operation related tests from `ibis.tests.exprs` to `ibis.expr.operations.tests`

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

File: ibis/common/patterns.py [hunk 1]
```diff
@@ -1596,7 +1596,10 @@ def pattern(obj: AnyType) -> Pattern:
     elif isinstance(obj, Mapping):
         return PatternMapping(obj)
     elif isinstance(obj, type):
-        return InstanceOf(obj)
+        if issubclass(obj, Coercible):
+            return CoercedTo(obj)
+        else:
+            return InstanceOf(obj)
```
Comment (cpcloud) [inline ibis/common/patterns.py:1602]: Is this related to testing? If not, can you submit a separate PR after you've moved things around?

Comment (kszucs) [reply ibis/common/patterns.py:1602]: I'm extracting it to another commit, but the changes in the test files depend on it.

File: ibis/expr/operations/tests/test_generic.py [hunk 2]
```diff
@@ -25,72 +25,52 @@ def test_literal_coercion_type_inference(value, dtype):
...
-    p = GenericCoercedTo(ops.Literal[dt.Int8])
-    assert p.match(ops.Literal(1, dt.int8), {}) == one
+    assert match(ops.Literal, 1) == one
+    assert match(ops.Literal, one) == one
+    assert match(ops.Literal, False) == ops.Literal(False, dt.boolean)
 
-    p = Pattern.from_typehint(ops.Literal[dt.Int8])
-    assert p == GenericCoercedTo(ops.Literal[dt.Int8])
+    assert match(ops.Literal[dt.Int8], 1) == one
+    assert match(ops.Literal[dt.Int16], 1) == ops.Literal(1, dt.int16)
```
Comment (cpcloud) [inline ibis/expr/operations/tests/test_generic.py:35]: Can you avoid changing a bunch of code in a PR that's nominally only about reorganizing tests?

Comment (kszucs) [reply ibis/expr/operations/tests/test_generic.py:35]: Yes, I thought that it worth some simplification once I go trough the test cases.

Comment (kszucs) [reply ibis/expr/operations/tests/test_generic.py:35]: All those changes are semantically identical with the previous versions, just shorter in length and easier to read.

Comment (cpcloud) [reply ibis/expr/operations/tests/test_generic.py:35]: Definitely in favor of the changes, just want to be able to give them the review they deserve!

Comment (kszucs) [reply ibis/expr/operations/tests/test_generic.py:35]: Would you like to see the changes in a separate commit or in a separate PR?

Comment (cpcloud) [reply ibis/expr/operations/tests/test_generic.py:35]: Separate commit is fine!

Comment (kszucs) [reply ibis/expr/operations/tests/test_generic.py:37]: Done.

**Review by cpcloud — APPROVED** [review]
